kms/capi/ncrypt_windows.go

```diff
@@ -289,6 +291,18 @@ func nCryptOpenKey(provisionerHandle uintptr, containerName string, legacyKeySpe
 	return kh, nil
 }
 
+func nCryptDeleteKey(keyHandle uintptr) error {
+	r, _, _ := procNCryptDeleteKey.Call(
+		keyHandle,
+		0,
+	)
+	if r != 0 {
+		return fmt.Errorf("NCryptDeleteKey returned %v", errNoToStr(uint32(r)))
+	}
```

Comment:
This might need a case similar to this before checking `r`:

```golang
if !errors.Is(err, windows.Errno(0)) {
    return nil, fmt.Errorf("NCryptDeleteKey returned %w", err)
}
```